solr/core/src/java/org/apache/solr/cloud/api/collections/AddReplicaCmd.java

```diff
@@ -144,7 +140,8 @@ public void call(ClusterState state, ZkNodeProps message, @SuppressWarnings({"ra
       }
     }
 
-    List<CreateReplica> createReplicas = buildReplicaPositions(ocmh.cloudManager, clusterState, collectionName, message, replicaTypesVsCount)
+    List<CreateReplica> createReplicas = buildReplicaPositions(ocmh.cloudManager, clusterState, collectionName, message, replicaTypesVsCount,
+        ocmh.overseer.getCoreContainer().getPlacementPluginFactory().createPluginInstance())
```

Comment:
Unclear to me: what happens here when placement plugins are not configured and we use for example the legacy assign strategy?

Reply:
Ok, I see that's how we tell which one is configured by this value being `null`... Can't say I really like it, but at least it should be commented (here and in `PlacementPluginFactory` as well?).

Reply:
The plugin is null and `Assign.createAssignStrategy` provides a `LegacyAssignStrategy`.